src/html/angular-i18n-translations-extractor.ts

```diff
@@ -0,0 +1,108 @@
+import {AngularElement, HtmlTranslationExtractionContext} from "./html-translation-extractor";
+import {Attribute} from "./element-context";
+
+const I18N_ATTRIBUTE_REGEX = /^i18n-.*$/;
+const I18N_ATTRIBUTE_NAME = "i18n";
+const ID_INDICATOR = "@@";
+
+/**
+ * Angular uses i18n and i18n-[attr] attributes for internationalization.
+ * The angularI18nTranslationsExtractor looks for these attributes on elements
+ * and extracts the found ids and default translations from the elements.
+ *
+ * @example
+ * <div i18n="@@translationId">some translation</div>
+ * results in a translation with id: 'translationId' and default translation: 'some translation'
+ *
+ * @example
+ * <div i18n-title="@@titleId" title="some title"></div>
+ * results in a translation with id: 'titleId' and default translation: 'some title'
+ *
+ * @param element the element to check for translations
+ * @param context the current context
+ */
+export default function angularI18nTranslationsExtractor(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext
+): void {
+    const i18nElementTranslation = element.attributes.find(attribute => attribute.name === I18N_ATTRIBUTE_NAME);
+
+    if (i18nElementTranslation) {
+        handleTranslationsOfElements(element, context, i18nElementTranslation);
+    }
+
+    const i18nAttributeTranslation = element.attributes.filter(attribute => I18N_ATTRIBUTE_REGEX.test(attribute.name));
+
+    handleTranslationsOfAttributes(element, context, i18nAttributeTranslation);
+}
+
+function handleTranslationsOfElements(
+    element: AngularElement,
+    context: HtmlTranslationExtractionContext,
+    attribute: Attribute
+): void {
+    const translationId = extractTranslationId(attribute, context);
+
+    if (element.texts.length > 0 && translationId) {
```

Comment:
I changed the implementation of the default handler for the case where an element has multiple child text nodes. 

```html
<element> 
 This is the first text child node, <span>I'm a child element</span>, and I'm the second text child node.
</element>
```